### PR TITLE
Tweak: Use user language in Elementor [ED-6556]

### PR DESCRIPTION
**Pull request by StyleShit**

### CODE REVIEW

File: includes/managers/widgets.php [hunk 1]
```diff
@@ -77,7 +78,7 @@ private function init_widgets() {
 		$this->_widget_types = [];
 
 		foreach ( $build_widgets_filename as $widget_filename ) {
-			include( ELEMENTOR_PATH . 'includes/widgets/' . $widget_filename . '.php' );
+			require_once ELEMENTOR_PATH . 'includes/widgets/' . $widget_filename . '.php';
```
Comment (matipojo) [inline includes/managers/widgets.php:81]: Revert

File: includes/managers/widgets.php [hunk 2]
```diff
@@ -189,7 +190,18 @@ private function register_wp_widgets() {
 	 * @access private
 	*/
 	private function require_files() {
-		require ELEMENTOR_PATH . 'includes/base/widget-base.php';
+		require_once ELEMENTOR_PATH . 'includes/base/widget-base.php';
```
Comment (matipojo) [inline includes/managers/widgets.php:193]: Revert

**Review by matipojo — CHANGES_REQUESTED** [review]

File: includes/managers/widgets.php (hunk 1 above)
Comment (StyleShit) [reply includes/managers/widgets.php:81]: We still need that

File: includes/managers/widgets.php (hunk 2 above)
Comment (StyleShit) [reply includes/managers/widgets.php:193]: Same as above

File: tests/phpunit/elementor/includes/managers/test-widgets.php [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+		// Assert.
+		// Button has translated defaults while inner-section doesn't.
+		$this->assert_array_have_keys( [ 'button' ], $response );
+		$this->assert_array_not_have_keys( [ 'inner-section' ], $response );
+	}
+
+	private function set_user_locale( $locale ) {
+		update_user_meta( get_current_user_id(), 'user_lang', $locale );
+	}
```
[the remark this answers is not in the document]
Comment (StyleShit) [reply tests/phpunit/elementor/includes/managers/test-widgets.php:70]: Nice catch

File: includes/compatibility.php [hunk 4]
```diff
@@ -238,26 +238,6 @@ public static function filter_library_post_type( $post_types ) {
 	 * @static
 	 */
 	private static function polylang_compatibility() {
-		// Fix language if the `get_user_locale` is difference from the `get_locale
```
[the remark this answers is not in the document]
Comment (StyleShit) [reply includes/compatibility.php:241]: Let's talk abotu it

File: tests/bootstrap.php [hunk 5]
```diff
@@ -71,3 +73,22 @@
...
+ */
+function copy_language_files( $_tests_dir ) {
+	$tests_data_dir = $_tests_dir . '/data';
+	$tests_lang_dir = $tests_data_dir . '/languages';
+
+	@mkdir( $tests_data_dir );
+	@mkdir( $tests_lang_dir );
+	@mkdir( $tests_lang_dir . '/plugins' );
+
+	touch( $tests_lang_dir . '/he_IL.mo' );
```
[the remark this answers is not in the document]
Comment (StyleShit) [reply tests/bootstrap.php:92]: ?

File: includes/managers/widgets.php (hunk 1 above)
Comment (matipojo) [reply includes/managers/widgets.php:81]: Nope

File: tests/bootstrap.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (matipojo) [reply tests/bootstrap.php:92]: !

**Pull request merged**